app/src/main/java/com/naman14/timberx/ui/activities/MainActivity.kt

```diff
@@ -69,8 +71,62 @@ class MainActivity : AppCompatActivity() {
         setupUI()
     }
 
-    private fun setupUI() {
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
```

Comment:
One of my libraries ([Assent](https://github.com/afollestad/assent)) could cleanup permissions code in this app a lot 😏

Reply:
Permissions code is only at one place currently so its not that problematic, (though Ideally it should also be checking for permission in onResume)